docs/development/module/fees.md

```diff
@@ -0,0 +1,193 @@
+---
+title: Transaction Weight and Fees
+---
+
+When transactions are submitted to a blockchain, they are executed by the nodes
+in the network. To be economically sustainable, the cost nodes incur to execute
+a transaction must be covered by the submitter of the transaction. The cost to
+execute transactions could vary over orders of magnitude, and thus Substrate
+provides a flexible mechanism for characterizing the total, presumably minimum,
+cost of a transaction in order to be included into a block.
+
+Furthermore, a number of resources in any chain can be limited by definition
+such as time, memory and computation. A mechanism should exist to prevent
+individual resource-consuming components of the chain, including but not limited
+to dispatchables, to consume too much of any of the mentioned resources. This
+concept is represented in substrate by weights.
+
+## Weights Basics
+
+The fee system is dependent on the
+[weight](https://crates.parity.io/sr_primitives/weights/index.html) system.
+Hence, we will first briefly describe how weights work.
+
+As mentioned, weights can technically refer to, or represent, numerous _limited_
+resources. A custom implementation may use complex structures to demonstrate
+this. At the time of this writing, substrate weights are simply a [numeric
+value](https://crates.parity.io/sr_primitives/weights/type.Weight.html). Each
+dispatchable function is given a weight using the `#[weight = $x]` annotation,
+where `$x` is some function capable of determining the weight of the dispatch.
+`$x` can access and examine the arguments of the dispatch. Nonetheless, a weight
+calculation should always:
+- Be computable __ahead of dispatch__. A block producer, or a validator, should
+  be able to examine the weight of a dispatchable before actually deciding to
+  accept it or not.
+- While not enforced, calculating the weight should not consume much resource
+  itself. It does not make much sense to spend almost the same amount of time as
+  dispatching a transaction to estimate how much weight it will consume. Thus,
+  weight computation should typically be very fast, much faster than
+  dispatching. This is analogous to a `poll()` function of the `Future` trait.
```

Comment:
I am not really a fan of bringing up `poll` and future here. I think your statement is clear without this example.